src/sender.rs

```diff
@@ -931,12 +932,12 @@ where
     /// When no session with the recipient exists, we need to create one.
     #[tracing::instrument(
         level = "trace",
-        skip(self, unidentified_access, content, recipient),
-        fields(unidentified_access = unidentified_access.is_some(), recipient = %recipient),
+        skip(self, unidentified_access, content),
+        fields(unidentified_access = unidentified_access.is_some(), recipient = recipient.service_id_string()),
```

Comment:
```suggestion
        skip(self, unidentified_access, content, recipient),
        fields(unidentified_access = unidentified_access.is_some(), recipient = %recipient.service_id_string()),
```

Shouldn't we skip the recipient in order to replace it?

Reply:
Looks like you don't have to, `tracing` doesn't complain about `recipient` not being `fmt::Display`.

Reply:
> Note that overlap between the names of fields and (non-skipped) arguments will result in a compile error.

https://docs.rs/tracing/latest/tracing/attr.instrument.html#adding-fields

this is weeeeeird.

Reply:
Well, it doesn't result in a compile error, so imo we can ship it 🤷